### PR TITLE
convert client-side to use typescript

### PR DESCRIPTION
**Pull request by jozzhart**
- refactor small existing function 'expanding-textbox'
- refactor unit test
- update build
- add basic type definition for GOVUKFrontend

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,9 +14,9 @@
     "build": "yarn build-sass && yarn build-js && yarn copy-assets",
     "build-sass": "node-sass app/assets/sass/application.scss public/css/main.css --output-style compressed",
     "copy-assets": "cp -R ./node_modules/govuk-frontend/assets ./public/govuk-frontend",
-    "build-js": "browserify ./app/assets/javascript/main.js -o ./public/js/main.js",
+    "build-js": "browserify ./app/assets/javascript/main.ts -p [ tsify --noImplicitAny ] > ./public/js/main.js",
     "test": "yarn test:unit && yarn test:browser",
-    "test:unit": "cross-env NODE_PATH=. NODE_ENV=test mocha test/unit --recursive",
+    "test:unit": "cross-env NODE_PATH=. NODE_ENV=test mocha  -r ts-node/register test/unit/**/*.test.ts --recursive",
```
Comment (davet1985) [inline package.json]: Doing this means that we are not running the JS unit tests, might need to have a separate script for running the typescript ones or something?

File: package.json [hunk 2]
```diff
@@ -57,7 +57,10 @@
   },
   "devDependencies": {
     "@hmcts/eslint-config": "^1.4.0",
-    "browserify": "^16.2.2",
```
Comment (davet1985) [inline package.json]: Guessing this needs to come back now that you are using `browserify`?

Comment (jozzhart) [reply package.json]: indded

File: package.json (hunk 1 above)
Comment (davet1985) [reply package.json]: 👍 totally agree on the aim, but it might take a bit of time.

**Pull request merged**